src/Microsoft.OData.Client/Serialization/HttpClientRequestMessage.cs

```diff
@@ -89,6 +91,24 @@ public HttpClientRequestMessage(DataServiceClientRequestMessageArgs args)
             }
         }
 
+#if NETSTANDARD2_0_OR_GREATER
+        /// <summary>
+        /// Gets the underlying <see cref="HttpClientHandler.ServerCertificateCustomValidationCallback"/> so that callers can customize their certificate valiation
+        /// during OData requests
+        /// </summary>
+        public Func<HttpRequestMessage, X509Certificate2, X509Chain, SslPolicyErrors, bool> ServerCertificateCustomValidationCallback

```

Comment:
Can we wrap those input/required parameters together as a CertificateValidationContext?
Then, it could be more clear to use.
And We can add more into the context later and without breaking anything?

Reply:
We are just surfacing the property of the underlying handler; seems like it would be confusing to change how the parameters were specified, and would require them to have a different func than they use for custom certificate validation in other places.

Reply:
Agreed with Mike here. @xuzhg, maybe this can be part of our "long-term 7.x" discussion, but I think here we should maintain the contract set out by `HttpClientHandler`. Is that alright for you?

Reply:
Emm... alright.